CRM/Core/Transaction.php

```diff
@@ -248,6 +248,7 @@ static public function isActive() {
    */
   static public function addCallback($phase, $callback, $params = NULL, $id = NULL) {
     $frame = \Civi\Core\Transaction\Manager::singleton()->getBaseFrame();
+    if (empty($frame)) return NULL;
```

Comment:
This makes me a little nervous -- the caller expects the callback to run, but it will never run. IMHO, generating an error is the right thing to do there.

After doing some grepping, it looks addCallback() is called by Civi\CCase\Events::fireCaseChange() -- which doesn't do anything to ensure that the transaction is active. It just assumes that transactions are available. Is that part of the backtrace when you encounter the error?

Perhaps this could be fixed with one or both of these:
 * Make sure that the "Change Timeline" business logic runs in a transaction by calling "$tx = new CRM_CoreTransaction()" or "CRM_Core_Transaction::create()->run(...);" in that business logic.
 * Make sure that fireCaseChange() runs in a transaction by calling "$tx = new CRM_Core_Transaction()" or "CRM_Core_Transaction::create()->run(...)".

For more discussion of how to ensure that a transaction is active: http://wiki.civicrm.org/confluence/display/CRMDOC/Transaction+Reference